app/authors/CreateAuthorForm.js

```diff
@@ -0,0 +1,132 @@
+"use client";
+
+import { gql, useMutation } from "@apollo/client";
+import { useRouter } from "next/navigation";
+import { useEffect, useState, useTransition } from "react";
```

Comment:
you can think of using a form component, lot of good libraries will help you in managing your form and form data.